x/incentives/keeper/distribute_test.go

```diff
@@ -545,62 +543,102 @@ func (suite *KeeperTestSuite) TestNoLockNonPerpetualGaugeDistribution() {
+			gaugeId:          validGaugeId,
+			expectedPoolType: poolmanagertypes.Concentrated,
+			expectErr:        false,
 		},
 		{
-			name:      "Invalid gaugeId and poolType",
-			gaugeId:   2,
-			expectErr: true,
+			name:                 "valid gaugeId and pool id link with balancer pool",
+			gaugeId:              validGaugeId,
+			expectedPoolType:     poolmanagertypes.Balancer,
+			shouldSetPoolGaugeId: true,
+			expectErr:            false,
+		},
+		{
+			name:                 "invalid gaugeId and pool id link and balancer pool",
+			gaugeId:              validGaugeId,
+			expectedPoolType:     poolmanagertypes.Balancer,
+			shouldSetPoolGaugeId: false,
+			expectErr:            true,
+		},
+		{
+			name:                    "valid gaugeId and pool id link with concentrated pool",
+			gaugeId:                 validGaugeId,
+			expectedPoolType:        poolmanagertypes.Concentrated,
+			shouldAvoidCreatingPool: true,
+			expectErr:               true,
 		},
 		{
-			name:              "Valid gaugeId, invalid poolType",
-			gaugeId:           1,
-			expectedGaugeType: poolmanagertypes.Balancer,
-			expectErr:         false,
+			name:      "invalid gaugeId and poolType",
+			gaugeId:   2,
+			expectErr: true,
 		},
 	}
 
 	for _, tc := range tests {
 		suite.Run(tc.name, func() {
 			suite.SetupTest()
 
-			suite.PrepareConcentratedPool()
 			incParams := suite.App.IncentivesKeeper.GetEpochInfo(suite.Ctx)
+			duration := incParams.GetDuration()
+
+			if !tc.shouldAvoidCreatingPool {
+				if tc.expectedPoolType == poolmanagertypes.Concentrated {
+					suite.PrepareConcentratedPool()
+				} else {
+					suite.PrepareBalancerPool()
+				}
+			}
+
+			if tc.shouldSetPoolGaugeId {
+				suite.App.PoolIncentivesKeeper.SetPoolGaugeId(suite.Ctx, validPoolId, duration, validGaugeId)
+			}
 
-			_, isClPool := suite.App.IncentivesKeeper.GetValidConcentratedLiquidityGauge(suite.Ctx, tc.gaugeId, incParams.Duration)
+			pool, err := suite.App.IncentivesKeeper.GetPoolFromGauge(suite.Ctx, tc.gaugeId, duration)
 			if tc.expectErr {
-				suite.Require().False(isClPool)
+				suite.Require().Error(err)
+				suite.Require().Nil(pool)
 			} else {
-				suite.Require().True(isClPool)
+				suite.Require().NoError(err)
+				suite.Require().NotNil(pool)
+				suite.Require().Equal(pool.GetType(), tc.expectedPoolType)
 			}
 		})
 	}
 }
 
-func (suite *KeeperTestSuite) TestDistributeConcentratedLiquidityInternal() {
-	defaultGauge := perpGaugeDesc{
-		lockDenom:    defaultLPDenom,
-		lockDuration: defaultLockDuration,
-		rewardAmount: sdk.Coins{sdk.NewInt64Coin(defaultRewardDenom, 3000)},
-	}
-	doubleLengthGauge := perpGaugeDesc{
-		lockDenom:    defaultLPDenom,
-		lockDuration: 2 * defaultLockDuration,
-		rewardAmount: sdk.Coins{sdk.NewInt64Coin(defaultRewardDenom, 3000)},
-	}
-
-	defaultGaugeStartTime := suite.Ctx.BlockTime()
+func (suite *KeeperTestSuite) TestDistributeConcentratedLiquidity() {
```

Comment:
Note: this test was not running so made minimal changes needed to let it run and pass